src/osvr/Client/SkeletonRemoteFactory.cpp

```diff
@@ -0,0 +1,152 @@
+                util::time::TimeValue const &timestamp) {
+                m_handleSkeletonSpec(data, timestamp);
+            });
+
+        OSVR_DEV_VERBOSE("Constructed a Skeleton Handler for " << deviceName);
+    }
+
+    void SkeletonRemoteHandler::m_handleSkeleton(
+        common::SkeletonNotification const &data,
+        util::time::TimeValue const &timestamp) {
+        if (*m_sensor != data.sensor) {
+            /// doesn't match our filter.
+            return;
+        }
+
+        if (m_skeletonConf) {
+            // get the latest joint and bone states
+            m_skeletonConf->updateSkeletonPoses();
+            // send skeleton report
+            OSVR_SkeletonReport report;
+            report.sensor = data.sensor;
+            report.state.skeleton = m_skeletonConf.get();
+            m_internals.setStateAndTriggerCallbacks(timestamp, report);
+        }
+    }
+
+    void SkeletonRemoteHandler::m_handleSkeletonSpec(
+        common::SkeletonSpec const &data,
+        util::time::TimeValue const &timestamp) {
+        // need to verify that sensor id and spec exist,
+        // otherwise don't get anything
+        if (!data.spec) {
+            return;
+        }
+        // get the articulation spec for specified skeleton sensor
+        Json::Value articSpec = data.spec[(*m_sensor)];
+        // update only if there are changes to articulation spec
+        if (m_articulationSpec != articSpec) {
+            m_articulationSpec = data.spec[(*m_sensor)];
+            osvr::common::PathTree articulationTree;
+            articulationTree.reset();
+            osvr::common::processArticulationSpecForPathTree(
+                articulationTree, m_deviceName, articSpec);
+            // skeleton config exists and needs to be updated
+            if (m_skeletonConf) {
+                m_skeletonConf->updateArticulationSpec(articulationTree);
+            } else {
+                try {
+                    m_skeletonConf.reset(
+                        new OSVR_SkeletonObject(m_ctx, articulationTree));
+                } catch (std::exception &e) {
+                    OSVR_DEV_VERBOSE(
+                        "Error creating skeleton object : constructor "
+                        "threw exception :"
+                        << e.what());
+                } catch (...) {
+                    OSVR_DEV_VERBOSE(
+                        "Error creating skeleton object : unknown exception");
+                }
+            }
+        }
+    }
+
+    SkeletonRemoteFactory::SkeletonRemoteFactory(
+        VRPNConnectionCollection const &conns)
+        : m_conns(conns) {}
+
+    shared_ptr<RemoteHandler> SkeletonRemoteFactory::
+    operator()(common::OriginalSource const &source,
+               common::InterfaceList &ifaces, common::ClientContext &ctx) {
+
+        shared_ptr<RemoteHandler> ret;
+
+        auto const &devElt = source.getDeviceElement();
+
+        /// @todo find out why make_shared causes a crash here
```

Comment:
Leftover `@todo` from before. Other RemoteFactories have the same comment. That's something needs to be addressed outside this PR